paimon-common/src/main/java/org/apache/paimon/catalog/CatalogContext.java

```diff
@@ -42,14 +46,17 @@ public class CatalogContext {
     private final Options options;
     private final Configuration hadoopConf;
     @Nullable private final FileIOLoader fallbackIOLoader;
+    @Nullable private final LineageMeta lineageMeta;
 
     private CatalogContext(
             Options options,
             @Nullable Configuration hadoopConf,
-            @Nullable FileIOLoader fallbackIOLoader) {
+            @Nullable FileIOLoader fallbackIOLoader,
+            @Nullable LineageMeta lineageMeta) {
```

Comment:
It looks like `LineageMeta` should be together with `LockFactory`. They are created by catalog. Not to create catalog? (CatalogContext is for creating catalog)

Reply:
@JingsongLi I have updated this PR to create lineage meta in AbstractCatalog